### DISCUSSION TITLE
Start G-code question & messed up max/min positions

### DISCUSSION THREAD

**Discussion by samuelT2** [mriscoc/Ender3V2S1#769] [Q&A] [answered]
Hi all

@mriscoc , this is by far one of the most helpful and well documented repo I have found so far for Merlin, thank you very much for your hard work, much appreciated!

I flashed my new E3 V2 Neo with the latest firmware release from mriscoc. So far so good. Still I'm struggling with some issues which I wasn't able to solve by myself yet:

1. What start G-code triggers a mesh calibration and uses it straight away, rather then a stored mesh? My start G-code below is my current one. I'd like to modify it to always trigger a mesh calibration.
2. Somehow, my max postions are messed up. I understand that I can set them by myself and tried to follow the calibration guide [found in the wiki](https://github.com/mriscoc/Ender3V2S1/wiki/Calibration-Guides#finding-the-minmax-position). But there I'm stuck on step 8 since I cannot find these menu entries in "Advanced settings". For the probe XY offset I used the coordinates given in [this issue](https://github.com/mriscoc/Ender3V2S1/issues/615). All my prints are way too lefty, the probe even sits straight on the left edge, whilst it has more then plenty of space on the right. My physical settings are:
```
X bed size: 220
y bed size: 220
X min pos: 20
Y min pos: 0
X max pos: 255
Y max pos: 231
Z max pos: 250
```
I guess those numbers are correct, given I have measured them by moving the axis to the max by hand. What else am I missing? I guess a calibration step. Even when I move the axis as proposed to 110/110, the nozzle is not in the middle at all.

Please note that I'm a newbie within the mriscoc firmware (only a couple of months into 3D printing at all) and might have misunderstood some concepts or hints.

Thank you very much for your help.

Start G-code:
```
G90 ; use absolute coordinates
M83 ; extruder relative mode
M104 S150 ; set temporary nozzle temp to prevent oozing during homing and auto bed leveling
M140 S{first_layer_bed_temperature[0]} ; set final bed temp
G4 S30 ; allow partial nozzle warmup
G28 ; home all axis
G29 L0 ; load a valid mesh from slot 0
G29 J ; Probe 3 points to tilt mesh if needed
G29 A  ; active the UBL system
G1 Z50 F240
G1 X2.0 Y10 F3000
M104 S{first_layer_temperature[0]} ; set final nozzle temp
M190 S{first_layer_bed_temperature[0]} ; wait for bed temp to stabilize
M109 S{first_layer_temperature[0]} ; wait for nozzle temp to stabilize
G1 Z0.28 F240
G92 E0
G1 X2.0 Y140 E10 F1500 ; prime the nozzle
G1 X2.3 Y140 F5000
G92 E0
G1 X2.3 Y10 E10 F1200 ; prime the nozzle
G92 E0
```

**Comment by samuelT2** [comment]
I somehow missed the discussion [here ](https://github.com/mriscoc/Ender3V2S1/discussions/413)where some physical settings are provided by other users. I'll try them all out first :). Though, I haven't found an answer to my first question yet.

**Comment by mriscoc** [comment]
Hi, thanks for your contribution.
- Some reports are indicating that the Marlin tilt mesh is not working as intended, remove it for now: G29 J
- For run an UBL auto leveling write G29 P1 as described at https://github.com/mriscoc/Ender3V2S1/wiki/Slicer-G-code-Scripts#optional
- Step 8 is inside of `Advanced / Mesh leveling / Mesh inset` submenu.

**Comment by samuelT2** [comment]
Thank you @mriscoc. I removed the G29 J and added G29 P1 as proposed. This looks fine so far, thanks. Within the calibration guide, I'm stuck now on step 9 where I want to connect Pronterface to the printer. Your [docu for Octoprint](https://github.com/mriscoc/Ender3V2S1/wiki/Octoprint#high-speed-serial-communication) states that baud should be set to 250'000 - my adapter allows max 128000. Can I switch the firmware to 115200 instead or does that need recompilation?

![image](https://user-images.githubusercontent.com/10983143/222257515-d84fcc42-6e3f-447b-9868-8074c267b19b.png)

**Reply by mriscoc** [reply]
> Can I switch the firmware to 115200

Yes, you can change the baud rate, save settings after that to make it permanent.

**Comment by samuelT2** [comment, marked as answer]
Nice, the switch to baud 115200 worked and I was able to connect to the printer. With that, I was also able to finish the calibration guide - though, my problem stays and the prints are not centered. I'll follow the calibration guide again from start to end.

**Reply by mriscoc** [reply]
If you follow the calibration guide and moving the head to the center is ok then the problem could be in your slicer. in the slicer offset must be 0 and check the size of the bed.

**Reply by samuelT2** [reply]
Indeed. PrusaSlicer has reset my plate back to 220x220 instead of 230x230. Now the printer prints more or less centered. Guess this is a fine-tuning problem now, to find my own physical settings. Thank you though, that was ride... I got lost in all these settings.
